src/app/Components/product-modal/product-modal.component.ts

```diff
@@ -8,13 +9,23 @@ import { MAT_DIALOG_DATA } from '@angular/material/dialog';
 })
 export class ProductModalComponent implements OnInit {
 
-  constructor( @Inject(MAT_DIALOG_DATA) public data:any) { }
+  constructor( @Inject(MAT_DIALOG_DATA) public data:any,
+   private cartService:CartService,
+   private matDialogRef:MatDialogRef<ProductModalComponent>
+   ) { }
 
   ngOnInit(): void {
   }
 
+  closeModal(){
+    this.matDialogRef.close()
+  }
   check(){
     alert('pumpum')
   }
 
+  addToCart(item:any){
```

Comment:
Here, Remember that the type of the data that the cart requires, is a cartItem. 
Which requires: 
1. id
2. product
3. quantity
4. totalPrice

Therefore all these should be calculated before they are passed to the cartService method